### PR TITLE
pcsc-safenet: init at 10.0.37-0

### PR DESCRIPTION
**Pull request by wldhx**
###### Motivation for this change

A new take on #77786. Pcsc support for Aladdin / e5100 tokens, works for me with 0529:0620.

Closes #77786

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/pcsc-safenet/default.nix [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  autoPatchelfHook,
+  dpkg,
+  gtk2,
+  fetchurl,
+  lib,
+  openssl,
+  pcsclite,
+  stdenv
+}:
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/tools/security/pcsc-safenet/default.nix:12]:
```suggestion
{ stdenv
, lib
, fetchurl
, autoPatchelfHook
, dpkg
, gtk2
, openssl
, pcsclite
}:

stdenv.mkDerivation rec {
```

File: pkgs/tools/security/pcsc-safenet/default.nix [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+  pname = "pcsc-safenet";
+  version = "10.0.37-0";
+
+  # https://aur.archlinux.org/packages/sac-core/
+  src = fetchurl {
+    url = "https://storage.spidlas.cz/public/soft/safenet/SafenetAuthenticationClient-core-10.0.37-0_amd64.deb";
```
Comment (SuperSandro2000) [inline pkgs/tools/security/pcsc-safenet/default.nix:19]:
```suggestion
    url = "https://storage.spidlas.cz/public/soft/safenet/SafenetAuthenticationClient-core${version}_amd64.deb";
```

File: pkgs/tools/security/pcsc-safenet/default.nix [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    sha256 = "1r9739bhal7ramj1rpawaqvik45xbs1c756l1da96din638gzy5l";
+  };
+
+  dontBuild = true;
+  dontConfigure = true;
+
+  unpackPhase = ''
+    ${dpkg}/bin/dpkg-deb -x $src .
+    find .
+  '';
```
Comment (SuperSandro2000) [inline pkgs/tools/security/pcsc-safenet/default.nix:29]:
```suggestion
    dpkg-deb -x $src .
  '';
```

File: pkgs/tools/security/pcsc-safenet/default.nix [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    openssl
+    pcsclite
+  ];
+
+  runtimeDependencies = [
+    openssl
+  ];
+
+  nativeBuildInputs = [
+    autoPatchelfHook
```
Comment (SuperSandro2000) [inline pkgs/tools/security/pcsc-safenet/default.nix:42]:
```suggestion
    autoPatchelfHook
    dpkg
```

File: pkgs/tools/security/pcsc-safenet/default.nix [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    cd "$out/lib/"
+    ln -sf libeToken.so.10.0.37 libeTPkcs11.so
+    ln -sf libeToken.so.10.0.37 libeToken.so.10.0
+    ln -sf libeToken.so.10.0.37 libeToken.so.10
+    ln -sf libeToken.so.10.0.37 libeToken.so
+    ln -sf libcardosTokenEngine.so.10.0.37 libcardosTokenEngine.so.10.0
+    ln -sf libcardosTokenEngine.so.10.0.37 libcardosTokenEngine.so.10
+    ln -sf libcardosTokenEngine.so.10.0.37 libcardosTokenEngine.so
+
+    cd "$out/pcsc/drivers/aks-ifdh.bundle/Contents/Linux/"
```
Comment (SuperSandro2000) [inline pkgs/tools/security/pcsc-safenet/default.nix:70]:
```suggestion
    cd $out/pcsc/drivers/aks-ifdh.bundle/Contents/Linux/
```

File: pkgs/tools/security/pcsc-safenet/default.nix [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+      mod_rpath="$(patchelf --print-rpath "$mod")"
+      patchelf --set-rpath "$runtime_rpath:$mod_rpath" "$mod"
+    done;
+  '';
+
+  meta = with lib; {
+    homepage = "https://safenet.gemalto.com/multi-factor-authentication/security-applications/authentication-client-token-management";
+    description = "Safenet Authentication Client";
+    platforms = [ "x86_64-linux" ];
+    license = licenses.unfree;
```
Comment (SuperSandro2000) [inline pkgs/tools/security/pcsc-safenet/default.nix:95]: Missing maintainers.

File: pkgs/tools/security/pcsc-safenet/default.nix [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+  };
+}
+
```
Comment (SuperSandro2000) [inline pkgs/tools/security/pcsc-safenet/default.nix:98]: A final new line to much.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by wldhx** [comment]
@SuperSandro2000 This should be it, cheers!

File: pkgs/tools/security/pcsc-safenet/default.nix [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+, autoPatchelfHook
+, dpkg
+, gtk2
+, openssl
+, pcsclite
+}:
+
+stdenv.mkDerivation rec {
+
+  pname = "pcsc-safenet";
```
Comment (SuperSandro2000) [inline pkgs/tools/security/pcsc-safenet/default.nix:13]:
```suggestion
stdenv.mkDerivation rec {
  pname = "pcsc-safenet";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
You forgot to add the package to all-packages.

**Comment by witchof0x20** [comment]
Glad to see someone's picking this up. I forgot about this. I'll close mine once this is merged.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110193` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>pcsc-safenet</li>
</ul>
</details>

**Comment by Et7f3** [comment]
Where did you get the binary from I don't see other version on this site. AUR seems to be using another repo again. Can you upgrade it ?

**Pull request merged**